core/devmode/src/main/java/io/quarkus/dev/JavaCompilationProvider.java

```diff
@@ -42,7 +43,8 @@ public void compile(Set<File> filesToCompile, Context context) {
             throw new RuntimeException("No system java compiler provided");
         }
         DiagnosticCollector<JavaFileObject> diagnostics = new DiagnosticCollector<>();
-        try (StandardJavaFileManager fileManager = compiler.getStandardFileManager(diagnostics, null, null);) {
+        try (StandardJavaFileManager fileManager = compiler.getStandardFileManager(diagnostics, null,
+                StandardCharsets.UTF_8)) {
```

Comment:
I agree we should have a fix in Wednesday's release.

Could you promote this PR and also create a proper GH issue. And I think it would be interesting to create a draft PR with your Maven work so that we don't lose it.

Reply:
PR promoted. Regarding the GH issue, there's already #3467.